cirq-google/cirq_google/devices/grid_device.py

```diff
@@ -122,7 +207,24 @@ class GridDevice(cirq.Device):
         * Get a collection of approximate gate durations for every gate supported by the device.
         >>> device.metadata.gate_durations
 
-        TODO(#5050) Add compilation_target_gatesets example.
+        * Get a collection of valid CompilationTargetGatesets for the device, which can be used to
+          transform a circuit to one which only contains gates from a native target gateset
+          supported by the device.
+        >>> device.metadata.compilation_target_gatesets
+
+        * Assuming valid CompilationTargetGatesets exist for the device, select the first one and
+          use it to transform a circuit to one which only contains gates from a native target
+          gateset supported by the device.
+        >>> cirq.optimize_for_target_gateset(
+                circuit,
+                gateset=device.metadata.compilation_target_gatesets[0]
+            )
+
+    A note about CompilationTargetGatesets:
+
+    A circuit which contains `cirq.WaitGate`s will be dropped if it is transformed using
+    CompilationTargetGatesets generated by GridDevice. To better control circuit timing, insert
+    WaitGates after the circuit has been transformed.
```

Comment:
I think this note should go to the docstring of `compilation_target_gatesets` property. This is a specific detail of compilation targets we are using today, and should not be tied to the description of device.

Reply:
My understanding is that we want to call out `WaitGate` because Google devices all support this gate. Is this concern general enough to all devices to be kept in the `compilation_target_gateset` property in `cirq.GridDeviceMetadata`? If so, I would suggest to keep it in `CompilationTargetGateset` instead, since all `CompilationTargetGatesets` would have this issue. If not, I would keep it where it is now.

Also, if we were to keep this comment in cirq-core, are there other identity-like gates that may not be negligible that we should call out?

Reply:
I think this is mainly used by the Google devices and we don't need to add it to the `CompilationTargetGateset` class.

Reply:
Sounds good, will keep the docstring here then since GridDevice doesn't have a `compilation_target_gatesets` property.